wazo_confgend/asterisk.py

```diff
@@ -52,115 +54,54 @@ def _generate_conf_from_generator(self, config_generator):
-
-            elif item['var_name'] not in ['allow', 'disallow']:
-                print >> output, "%s = %s" % (item['var_name'], item['var_val'])
-
-            elif item['var_name'] == 'allow':
-                print >> output, 'disallow = all'
-                for c in item['var_val'].split(','):
-                    print >> output, 'allow = %s' % c
-
-        return output.getvalue()
-
-    def _gen_iax_trunk(self, trunk):
-        output = StringIO()
-
-        print >> output, "\n[%s]" % trunk.name
-
-        exclude_options = ('id', 'name', 'protocol', 'category', 'commented', 'disallow')
-        for k, v in trunk.all_options(exclude=exclude_options):
-            if v in (None, ''):
-                continue
-
-            if isinstance(v, unicode):
-                v = v.encode('utf8')
-
-            if k == 'allow':
-                print >> output, "disallow = all"
-                for c in v.split(','):
-                    print >> output, "allow = " + str(c)
-            else:
-                print >> output, k + " = ", v
-
-        return output.getvalue()
+        config_generator = IaxConf()
+        return self._generate_conf_from_generator(config_generator)
 
     def queueskills_conf(self):
-        """Generate queueskills.conf asterisk configuration file
-        """
-        options = StringIO()
+        """Generate queueskills.conf asterisk configuration file"""
+        output = StringIO()
+        ast_writer = AsteriskFileWriter(output)
 
-        agentid = None
+        agent_id = None
         for sk in asterisk_conf_dao.find_agent_queue_skills_settings():
-            if agentid != sk['id']:
-                print >> options, "\n[agent-%d]" % sk['id']
-                agentid = sk['id']
-
-            print >> options, "%s = %s" % (sk['name'], sk['weight'])
+            if agent_id != sk['id']:
+                ast_writer.write_section('agent-{:d}'.format(sk['id']))
+                agent_id = sk['id']
+            ast_writer.write_option(sk['name'], sk['weight'])
 
-        return options.getvalue()
+        return output.getvalue()
 
     def queueskillrules_conf(self):
-        """Generate queueskillrules.conf asterisk configuration file
-        """
-        options = StringIO()
+        """Generate queueskillrules.conf asterisk configuration file"""
+        output = StringIO()
+        ast_writer = AsteriskFileWriter(output)
 
         for r in asterisk_conf_dao.find_queue_skillrule_settings():
-            print >> options, "\n[skillrule-%s]" % r['id']
-
+            ast_writer.write_section('skillrule-{}'.format(r['id']))
             if 'rule' in r and r['rule'] is not None:
                 for rule in r['rule'].split(';'):
-                    print >> options, "rule = %s" % rule
+                    ast_writer.write_option('rule', rule)
 
-        return options.getvalue()
+        return output.getvalue()
 
     def queuerules_conf(self):
-        options = StringIO()
+        output = StringIO()
```

Comment:
Why not using AsteriskFileWriter for this section?

Reply:
I didn't use it, because it writes the whole line in multiple steps, but you're right, I should. I I'll just build the string for the option and use the `write_object_option` after of writing directly to the output.